src/livestatus.c

```diff
@@ -0,0 +1,468 @@
+#include <sys/types.h>
+#include <sys/un.h>
+
+#define IO_BUFFER_SIZE 4096
+#define LIVESTATUS_FIELD_SEP ';'
+#define LIVESTATUS_PLUGIN_NAME "livestatus"
+
+typedef uint64_t lcounter_t;
+
+static const char *config_keys[] = {"LivestatusSocketFile", "OnFailureMaxRetry",
+                                    "OnFailureBackOffSeconds"};
+static int config_keys_num = STATIC_ARRAY_SIZE(config_keys);
+
+struct livestatus_s {
+  char *socket_file;
+  int max_retry;
+  int backoff_sec;
+};
+typedef struct livestatus_s livestatus_t;
+
+#define LIVESTATUS_QUERY_COLUMNS                                               \
+  "cached_log_messages "                                                       \
+  "connections "                                                               \
+  "connections_rate "                                                          \
+  "forks "                                                                     \
+  "forks_rate "                                                                \
+  "host_checks "                                                               \
+  "host_checks_rate "                                                          \
+  "livecheck_overflows "                                                       \
+  "livecheck_overflows_rate "                                                  \
+  "livechecks "                                                                \
+  "livechecks_rate "                                                           \
+  "log_messages "                                                              \
+  "log_messages_rate "                                                         \
+  "neb_callbacks "                                                             \
+  "neb_callbacks_rate "                                                        \
+  "requests "                                                                  \
+  "requests_rate "                                                             \
+  "service_checks "                                                            \
+  "service_checks_rate"
+
+#define LIVESTATUS_EXPECTED_FIELDS_RESP_NB 19
+
+struct livestatus_status_s {
+  int cached_log_messages;
+  lcounter_t connections;
+  double connections_rate;
+  lcounter_t forks;
+  double forks_rate;
+  lcounter_t host_checks;
+  double host_checks_rate;
+  lcounter_t livecheck_overflows;
+  double livecheck_overflows_rate;
+  lcounter_t livechecks;
+  double livechecks_rate;
+  lcounter_t log_messages;
+  double log_messages_rate;
+  lcounter_t neb_callbacks;
+  double neb_callbacks_rate;
+  lcounter_t requests;
+  double requests_rate;
+  lcounter_t service_checks;
+  double service_checks_rate;
+};
+typedef struct livestatus_status_s livestatus_status_t;
+
+static livestatus_t livestatus_obj = {
+    .socket_file = "/var/cache/naemon/live",
+    .max_retry = 20,
+    .backoff_sec = 1,
+};
+
+livestatus_status_t c_to_livestatus_status(const char **fields) {
```

Comment:
Please pass a `size_t fields_num` and verify that the array is large enough here. Keep in mind that we're also trying to be defensive in face of future changes. In other words, we should add this check even if the below code (currently) guarantees that there are at least 19 fields.